jax/_src/numpy/polynomial.py

```diff
@@ -292,6 +295,25 @@ def polymul(a1, a2, *, trim_leading_zeros=False):
   val = convolve(a1, a2, mode='full')
   return val
 
+@_wraps(np.polydiv, lax_description=_LEADING_ZEROS_DOC)
+def polydiv(u, v, *, trim_leading_zeros=False):
+  _check_arraylike("polydiv", u, v)
+  u, v = _promote_dtypes_inexact(u, v)
+  w = u[0] + v[0] # calculate type
+  m = len(u) - 1
+  n = len(v) - 1
+  scale = 1. / v[0]
+  q = zeros(max(m - n + 1, 1), dtype = w.dtype) # force same dtype
+  r = u.astype(w.dtype)
+  for k in range(0, m-n+1):
+    d = scale * r[k]
+    q = q.at[k].set(d)
+    r = r.at[k:k+n+1].add(-d*v)
+  if trim_leading_zeros:
+    return q, trim_zeros(
+      where((~(abs(r)<sqrt(finfo(w.dtype).eps))).cumsum().astype(bool), r, 0), trim='f')
```

Comment:
I imagine looking at this line in a year and having a lot of trouble understanding the logic. I would factor this out into a `trim_zeros_tol(filt, tol)` utility function that accepts a tolerance, and then add a comment describing why you are passing in the square root of eps

Reply:
Could you please tell me where I should put the ```trim_zeros_tol(filt, tol)```?